bok_choy/page_object.py

```diff
@@ -372,6 +373,7 @@ def validate_url(cls, url):
                 # Valid URLs do not end with colons.
                 return False
         except ValueError:
+            LOGGER.warning(u"%s uses an invalid port", url)
```

Comment:
I wonder if we should add logging to all of the cases in which this function returns False. Where the function is called, it raises an exception if False that says  'Invalid' but is that helpful? What do you think? A lot of the time, the reasons for a URL are easy to spot, but this might be useful.